### DISCUSSION TITLE
Documenting cpp-adapter.cpp for Android

### DISCUSSION THREAD

**Discussion by ChrisLFieldsII** [reactwg/react-native-new-architecture#205] [Q&A] [answered]
I looked through the docs and didn't see a mention of the `cpp-adapter.cpp` file which it looks like is required for Android in my minimal experience.

It seems like this file is responsible for handling the interoperability between Java and CPP?

I'm assuming this is the code that runs for the Java `native` methods? 

Is there any place we can find more info on this?

**Comment by cortinico** [comment, marked as answer]
> I looked through the docs and didn't see a mention of the `cpp-adapter.cpp` file which it looks like is required for Android in my minimal experience.

I've never heard about this file.

Where have you found it mentioned? Do you have any reference?

**Reply by ChrisLFieldsII** [reply]
I was following the [cross platform guide](https://github.com/reactwg/react-native-new-architecture/blob/main/docs/turbo-modules-xplat.md) for building a turbo module

I attached a picture that shows the `cpp-adapter.cpp` in the `android` dir of the sample turbo module I created to play around with this stuff.

I believe that file got there b/c I used [create-react-native-library](https://github.com/callstack/react-native-builder-bob/tree/main/packages/create-react-native-library/templates/cpp-library/android) as the RN docs specify [here](https://reactnative.dev/docs/native-modules-setup).

If I do not implement a cpp function that was defined in the JavaScript spec in the `cpp-adapter.cpp` file, then I get a runtime error on Android b/c it can't find the native implementation.

This had me stomped for a bit until I stumbled upon [Oracle's JNI docs](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html) which kinda helped me figure out about the JNI types like `JNIEnv`, `jstring`, `jdouble`

I figured this was the place to have documentation on something like this, but maybe there is a better place or most likely I am just confused.

<img width="1840" alt="Screenshot 2024-08-02 at 11 59 23 PM" src="https://github.com/user-attachments/assets/15a322a4-0864-45bd-a542-d51074abe22f">

<img width="1728" alt="Screenshot 2024-08-03 at 12 14 25 AM" src="https://github.com/user-attachments/assets/d92a2b80-a1a9-4cc5-a184-a2b7c17865f2">

**Reply by cortinico** [reply]
You should open an issue against `create-react-native-library` as `cpp-adapter` is a file they add. That's also why it's not referenced in the guide you linked

**Reply by ChrisLFieldsII** [reply]
Noted. Sounds like a misunderstanding on my part. Thanks!
